tools/sigh.ts

```diff
@@ -422,6 +446,101 @@ function link(srcFiles: Iterable<string>): boolean {
   return success;
 }
 
+// Config for building wasm modules.
+type WasmConfig = {
+  [key: string]: {         // The target filename for the module.
+    manifest: string,      // The manifest to process; should be in the same dir as the json file
+    src: string[],         // The list of source files to compile (currently limited to one)
+    outDir: string,        // The output directory relative to project root; should be under 'build' for tests
+    linkManifest: boolean  // Whether to link the manifest file into outDir; should be true for tests
+  }
+};
+
+// TODO: detect old headers/wasm modules/manifests in cleanObsolete()
+function buildWasmModule(configFile: string, logCmd: boolean): boolean {
```

Comment:
It seems like this needs to be extended to build wasm from kotlin given a wasm.json config. 
Sounds like interesting work for a later PR

Reply:
Please correct me if I'm assuming wrong. 
At the very least, we can write test particle artifacts for Kotlin

Reply:
That's one of the aims here - to provide integration points for testing both(/all) flavours of wasm. I'm starting with C++ because I know how to drive that, but it's very much intended that this "build" system be extended to handle kotlin.